ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -289,8 +289,7 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery', 'utils'],
         vocabulary: vocab
       };
       var vocabularies = this.state.vocabularies;
-      vocabularies.push(newVocab);
-      this.setState({vocabularies: vocabularies});
+      this.setState({vocabularies: vocabularies.concat([newVocab])});
```

Comment:
It's discouraged in [their documentation](https://facebook.github.io/react/docs/component-api.html). It may give other pieces of code (like unit tests) a false impression on what the state is since setState may have been batched for later execution, but the change here is happening immediately.